data/modifications/pick_advance/dialog.lua

```diff
@@ -0,0 +1,168 @@
+-- << pickadvance_dialog
+
+pickadvance = {}
+local pickadvance = pickadvance
+local wesnoth = wesnoth
+local ipairs = ipairs
+local table = table
+local T = wesnoth.require("lua/helper.lua").set_wml_tag_metatable {}
+local translate = wesnoth.textdomain "wesnoth"
+
+local function filter_false(arr)
+	local result = {}
+	for _, v in ipairs(arr) do
+		if v ~= false then
+			result[#result + 1] = v
+		end
+	end
+	return result
+end
+
+
+function pickadvance.show_dialog_unsynchronized(advance_info, unit)
+-- dialog exit codes --
+  local help_code = -4
+  local reset_code = -3
+  local single_unit_code = -1
+  local all_units_code = 1
+--    
+	local label = "Plan advance:"
+
+	local unit_type_options = advance_info.type_advances
+	--print_as_json("advances for", unit.type, unit_type_options)
+	local options = {}
+	for _, ut in ipairs(unit_type_options) do
+		options[#options + 1] = wesnoth.unit_types[ut]
+	end
+	local show_images = true
+
+	local unit_override_one = (advance_info.unit_override or {})[2] == nil
+		and (advance_info.unit_override or {})[1] or nil
+	local game_override_one = (advance_info.game_override or {})[2] == nil
+		and (advance_info.game_override or {})[1] or nil
+
+	local description_row = T.row {
+		T.column { T.label { use_markup = true, label = label } },
+	}
+
+	local list_sub_row
```

Comment:
With `show_images` inlined (see above), this would also be simplified.

Reply:
Done.